src/client/Microsoft.Identity.Client/Internal/Logger/MsalLogger.cs

```diff
@@ -233,5 +234,15 @@ internal static string GetPiiScrubbedExceptionDetails(Exception ex)
 
             return sb.ToString();
         }
+
+        public DurationLogHelper LogBlockDuration(string measuredBlockName, LogLevel logLevel = LogLevel.Verbose)
+        {
+            return new DurationLogHelper(this, measuredBlockName, logLevel);
+        }
+
+        public DurationLogHelper LogMethodDuration(LogLevel logLevel = LogLevel.Verbose, [CallerMemberName] string methodName = null)
```

Comment:
Another C# trick ... [CallerMemberName] will populate the param (methodName) with the name of calling method. This is done by the compiler, not at runtime (reflection is very slow)

https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=netcore-3.1

CC @jennyf19 in case you'll want to add some perf logging to Identity.Web

Reply:
very cool

---
In reply to: [436182007](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1866#discussion_r436182007) [](ancestors = 436182007)

Reply:
That's cool. asp.net core has something similar for logging.

---
In reply to: [436182007](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1866#discussion_r436182007) [](ancestors = 436182007)